edge-modules/TwinTester/TwinOperator.cs

```diff
@@ -0,0 +1,354 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class TwinOperator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("TwinTester");
+        private readonly object operationLock = new Object();
+        private RegistryManager registryManager;
+        private ModuleClient moduleClient;
+        private AnalyzerClient analyzerClient;
+        private Storage storage;
+        private TwinState twinState;
+
+        public TwinOperator(RegistryManager registryManager, ModuleClient moduleClient, AnalyzerClient analyzerClient, Storage storage)
+        {
+            this.registryManager = registryManager;
+            this.moduleClient = moduleClient;
+            this.analyzerClient = analyzerClient;
+            this.storage = storage;
+            this.twinState = this.InitializeModuleTwin().Result;
```

Comment:
Is this OK? It seems like this is a hidden dependency. If anyone has other suggestions I'm all ears